### DISCUSSION TITLE
How to charge while considering SoC?

### DISCUSSION THREAD

**Discussion by cyasing** [pybamm-team/PyBaMM#2872] [Q&A] [answered]
I wanted to know how to design the experiment so that it charges and discharges until a specific SoC is reached, like in a battery management system.

**Comment by RuiheLi** [comment, marked as answer]
Hi~ As far as I am aware, PyBaMM cannot use SOC as termination condition yet. You can find more [here](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/experiments/experiment.py).

**Reply by Mrzhang-hub** [reply]
I think it is mearningful to add SOC as an alternative termination condition and hope it can be added to pybamm.   @brosaplanella  @tinosulzer

**Reply by redchang111** [reply]
Is there an alternative way to achieve this

**Reply by brosaplanella** [reply]
If you know how much capacity you want to introduce, you can do a constant current charge for a certain time so you introduce a certain capacity.
